src/github.com/matrix-org/dendrite/clientapi/routing/sendevent.go

```diff
@@ -35,35 +35,25 @@ type sendEventResponse struct {
 	EventID string `json:"event_id"`
 }
 
-// SendEvent implements:
-//   /rooms/{roomID}/send/{eventType}
-//   /rooms/{roomID}/send/{eventType}/{txnID}
-//   /rooms/{roomID}/state/{eventType}/{stateKey}
-func SendEvent(
+func generateSendEvent(
```

Comment:
Could you move this (helper) function after SendEvent ?
This pattern is usually followed in Dendrite (and hopefully other places ..)
Pattern being: placing the public function first then all it's helpers.

p.s. Inner me wants this function be named just `sendEvent`, but anything is fine. :P

Reply:
Of course I can move it. I left it on top for a better readable diff.
But I would leave it as `generateSendEvent` to avoid name clashing. I would assume `generateEvent` might arise in other contexts as well.